lib/charty/vector_adapters/arrow_adapter.rb

```diff
@@ -0,0 +1,130 @@
+module Charty
+  module VectorAdapters
+    class ArrowAdapter < BaseAdapter
+      VectorAdapters.register(:arrow, self)
+
+      include Enumerable
+      include NameSupport
+      include IndexSupport
+
+      def self.supported?(data)
+        (defined?(Arrow::Array) && data.is_a?(Arrow::Array)) ||
+          (defined?(Arrow::ChunkedArray) && data.is_a?(Arrow::ChunkedArray))
+      end
+
+      def initialize(data)
+        @data = check_data(data)
+        self.index = index || RangeIndex.new(0 ... length)
+      end
+
+      def size
+        @data.length
+      end
+
+      def where(mask)
+        mask = check_mask_vector(mask)
+        unless mask.data.is_a?(Arrow::BooleanArray)
+          mask = mask.to_a
+          mask = mask.collect {|integer| integer != 0 } if mask[0].is_a?(Integer)
+          mask = Arrow::BooleanArray.new(mask)
+        end
+        masked_data = @data.filter(mask)
+        masked_index = []
+        mask.to_a.each_with_index do |boolean, i|
+          masked_index << index[i] if boolean
+        end
+        Charty::Vector.new(masked_data, index: masked_index, name: name)
+      end
+
+      def boolean?
+        case @data
+        when Arrow::BooleanArray
+          true
+        when Arrow::ChunkedArray
+          @data.value_data_type.is_a?(Arrow::BooleanDataType)
+        else
+          false
+        end
+      end
+
+      def numeric?
+        case @data
+        when Arrow::NumericArray
+          true
+        when Arrow::ChunkedArray
+          @data.value_data_type.is_a?(Arrow::NumericDataType)
+        else
+          false
+        end
+      end
+
+      def categorical?
+        case @data
+        when Arrow::DictionaryArray
```

Comment:
Why not treat `StringArray` as categorical?

Reply:
OK.
I'll do it.

Reply:
Done.